tools/tpm2_nvlist.c

```diff
@@ -49,18 +50,32 @@ static void print_nv_public(TPM2B_NV_PUBLIC *nv_public) {
         LOG_ERR("Could not convert attributes to string form");
     }
 
-    printf("  {\n");
-    printf("\tHash algorithm(nameAlg):%d\n ", nv_public->nvPublic.nameAlg);
-    printf("\tattributes: %s(0x%X)\n ", attrs,
+    const char *alg = tpm2_alg_util_algtostr(nv_public->nvPublic.nameAlg);
+    if (!alg) {
+        LOG_ERR("Could not convert algorithm to string form");
+    }
+
+    tpm2_tool_output("  hash algorithm:\n");
+    tpm2_tool_output("    friendly: %s\n", alg);
+    tpm2_tool_output("    value: 0x%X\n",
+            nv_public->nvPublic.nameAlg);
+
+    tpm2_tool_output("  attributes:\n");
+    tpm2_tool_output("    friendly: %s\n", attrs);
+    tpm2_tool_output("    value: 0x%X\n",
             tpm2_util_ntoh_32(nv_public->nvPublic.attributes.val));
-    printf("\tThe size of the data area(dataSize):%d\n ",
-            nv_public->nvPublic.dataSize);
-    printf("\tAuthorization Policy for R/W/D: ");
-    int i;
+
+    tpm2_tool_output("  size: %d\n",
+               nv_public->nvPublic.dataSize);
+
+    tpm2_tool_output("  authorization policy: ");
```

Comment:
Can we only print this if `nv_public->nvPublic.authPolicy.size > 0` ? Or print `None` or something if `nv_public->nvPublic.authPolicy.size == 0`. Otherwise it feels as if the output got truncated.

Reply:
That's fine. My reasoning for the way it is is that one can always get the key in the yaml file and the value is null. However, it's just as easy to check if key exists. So ill do it that way and then it's less for a human to read as well.